plugins/rbac/src/hooks/useRoles.ts

```diff
@@ -107,10 +105,11 @@ export const useRoles = (
       catalogEntityReadPermissionResult,
     ],
   );
-  const loading = rolesLoading && policiesLoading;
+  const loading = !roles && !policies;
```

Comment:
@debsmita1 this won't work if there is an error while fetching roles/policies as they will remain undefined. One thing you can do here is that check for errors as well. Like this `const loading =  !rolesError && !policiesError && !roles && !policies`. But since we already have a loading value you can also maintain a state for that and its value will only be set to false for the first time loading becomes false. I have mentioned the same here https://github.com/janus-idp/backstage-plugins/pull/1057#discussion_r1459004637. This is similar to what you were doing in the component file.

Reply:
@divyanshiGupta I  have tested this when there is an error while fetching the roles/policies. When there is an error with roles or policies, they return an object. Which will set loading as `false`

Reply:
Okay then that should be fine. Also we show errors in the list right? What will happen in case of a network error?

Reply:
@divyanshiGupta  Earlier I was not handling network error scenarios nor was any error shown in the list page. I have updated the code and attached screenshots in the description. Thank you for pointing this out.